backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/init/ResilientClusterProviderTest.java

```diff
@@ -0,0 +1,145 @@
+                        .build()))
+                .isInstanceOf(IllegalStateException.class)
+                .hasStackTraceContaining("User james_testing has no CREATE permission on <all keyspaces> or any of its parents");
+        }
+
+        @Test
+        void initializationWithPrivilegedUserShouldCreateKeySpaceWhenNotExisted(DockerCassandra dockerCassandra) {
+            new ResilientClusterProvider(dockerCassandra.configurationBuilderForSuperUser()
+                .keyspace(KEYSPACE)
+                .createKeyspace()
+                .build());
+
+            assertThat(dockerCassandra.administrator()
+                    .keyspaceExist(KEYSPACE))
+                .isTrue();
+        }
+
+        @Test
+        void initializationShouldNotThrownWhenKeyspaceAlreadyExisted(DockerCassandra dockerCassandra) {
+            ClusterConfiguration configuration = dockerCassandra.configurationBuilder()
+                .keyspace(KEYSPACE)
+                .createKeyspace()
+                .build();
+            dockerCassandra.administrator()
+                .initializeKeyspace(KEYSPACE);
+
+            assertThatCode(() -> new ResilientClusterProvider(configuration))
+                .doesNotThrowAnyException();
+        }
+
+        @Test
+        void initializationShouldNotImpactToKeyspaceExistentWhenAlreadyExisted(DockerCassandra dockerCassandra) {
+            ClusterConfiguration configuration = dockerCassandra.configurationBuilder()
+                .keyspace(KEYSPACE)
+                .createKeyspace()
+                .build();
+            dockerCassandra.administrator()
+                .initializeKeyspace(KEYSPACE);
+
+            new ResilientClusterProvider(configuration);
+
+            assertThat(dockerCassandra.administrator()
+                    .keyspaceExist(KEYSPACE))
+                .isTrue();
+        }
+    }
+
+    @Nested
+    class WhenProhibitCreatingKeySpace {
+
+        @Test
+        void initializationShouldNotCreateWhenKeyspaceDoesntExist(DockerCassandra dockerCassandra) {
+            new ResilientClusterProvider(dockerCassandra.configurationBuilder()
+                .keyspace(KEYSPACE)
+                .build());
+
+            assertThat(dockerCassandra.administrator()
+                    .keyspaceExist(KEYSPACE))
+                .isFalse();
+        }
+
+        @Test
+        void initializationShouldNotThrownWhenKeyspaceAlreadyExisted(DockerCassandra dockerCassandra) {
+            ClusterConfiguration configuration = dockerCassandra.configurationBuilder()
+                .keyspace(KEYSPACE)
+                .build();
+            dockerCassandra.administrator()
+                .initializeKeyspace(KEYSPACE);
+
+            assertThatCode(() -> new ResilientClusterProvider(configuration))
+                .doesNotThrowAnyException();
+        }
+
+        @Test
+        void initializationShouldNotImpactToKeyspaceExistentWhenAlreadyExisted(DockerCassandra dockerCassandra) {
```

Comment:
initializationShouldNotImpactKeyspaceExistenceWhenItAlreadyExisted ?

Maybe the test should better reflect that ? There is nothing asserting the keyspace existed before the creation of the ResilientClusterProvider ?

Reply:
That lead me to another question ? Why instantiating a Provider do those kind of side effects ?

Reply:
>  Why instantiating a Provider do those kind of side effects ?

I don't know, the logic has already been there. You can say that the ResilientClusterProvider should provide a working Cluster. That requires the keyspace to be created. Physically, it's side effect, logically it's not :)

Reply:
> There is nothing asserting the keyspace existed before the creation of the ResilientClusterProvider 

That's true, but I feel not comfortable to put that check in every test. I'm imagining writing the BlobStoreContract and write the blobStore empty check every times before I wrote a test. They are similar right? 

Will extracting the cluster creation into a separated `@PostConstruct` be okay, then I'm able to test the case.

Reply:
> > Why instantiating a Provider do those kind of side effects ?
> 
> I don't know, the logic has already been there. You can say that the ResilientClusterProvider should provide a working Cluster. That requires the keyspace to be created. Physically, it's side effect, logically it's not :)

Why don't do it at the first call ?
It's still a side effect :) It was making the test harder to understand,
but it's not really the scope of this PR so i will not bother you too much about it.